src/readbin/fsi_benchmark.ml

```diff
@@ -0,0 +1,120 @@
+open Core_kernel.Std
+open Cmdliner
+open Bap.Std
+open Bap_plugins.Std
+
+type metrics = [
+  | `with_F
+  | `with_FN
+  | `with_FP
+  | `with_TP
+  | `with_prec
+  | `with_recl
+]
+
+type evaluation = {
+  true_positive: int;
+  false_negative: int;
+  false_positive: int;
+  prec: float;
+  recl: float;
+  f_05: float}
+
+let tool : string Term.t =
+  let doc = "The tool of the function start result that we are going to
+  evaluate. If user wants to evaluate bap-byteweight, one can use
+  \"bap-byteweight\". If user want to evaluate IDA, one can use
+  \"idaq\", \"idal\", \"idaq64\" or the specific path of IDA." in
+  Arg.(required & pos 0 (some string) (Some "bap-byteweight") & info [] ~docv:"tool" ~doc)
+
+let bin : string Term.t =
+  let doc = "The testing stripped binary." in
+  Arg.(required & pos 1 (some non_dir_file) None & info [] ~docv:"binary" ~doc)
+
+let truth : string Term.t =
+  let doc =
+    "The ground truth. The ground truth can be an unstripped
+  binary, or a .scm file with symbol information. In .scm file, each symbol should
+  be in format of:
+  (<symbol name> <symbol start address> <symbol end address>), e.g.
+  (malloc@@GLIBC_2.4 0x11034 0x11038)" in
+  Arg.(required & pos 2 (some non_dir_file) None
+       & info [] ~docv:"ground-truth" ~doc)
+
+let print_metrics : metrics list Term.t =
+  let opts = [
+    "precision", `with_prec;
+    "recall", `with_recl;
+    "F_measure", `with_F;
+    "TP", `with_TP;
+    "FP", `with_FP;
+    "FN", `with_FN;
+  ] in
+  let doc = "Print metrics. User can choose to print -cprecision, -crecall,
+  -cF_measure, -cTP, -cFP and -cFN." in
+  Arg.(value & opt_all (enum opts) (List.map ~f:snd opts) &
+       info ["with-metrics"; "c"] ~doc)
+
+let output_metric_value oc {false_positive;false_negative;
```

Comment:
field value can be accessed with dot syntax, e.g., `r.false_positive`. 
Please use dot syntax here, it is much more concise.